### PR TITLE
Add nicer error for breeze update static check

### LINKED ISSUE #22908: Breeze: Add nicer error for BREEZE.rst update static check
If you modify BREEZE.rst or any of the "dev" files there is a static check run that updates screenshots in the docs.

If you have no Breeze installed, It should nicely inform the user that they have to install Breeze and redirect them to the documentation on how to do it.

But currently, it will fail with this message:

```
Update output of breeze commands in BREEZE.rst...........................................Failed
- hook id: update-breeze-file
- exit code: 1

Traceback (most recent call last):
  File "./scripts/ci/pre_commit/pre_commit_breeze_cmd_line.py", line 81, in <module>
    print_help_for_all_commands()
  File "./scripts/ci/pre_commit/pre_commit_breeze_cmd_line.py", line 46, in print_help_for_all_commands
    check_call(["breeze", "--help"], env=env)
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/subprocess.py", line 358, in check_call
    retcode = call(*popenargs, **kwargs)
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/subprocess.py", line 339, in call
    with Popen(*popenargs, **kwargs) as p:
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/subprocess.py", line 800, in __init__
    restore_signals, start_new_session)
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/subprocess.py", line 1551, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'breeze': 'breeze'
````

This should be fixed.

### PR DESCRIPTION
**Pull request by edithturn**
Closes #22908

Steps:
- First: Reproduced the error, modifying BREEZE.rst, It will update the output of breeze commands
- In pre_commit_breeze_cmd_line.py added verify_breeze_installed to verify if the breeze is installed. If not print a message.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Comment by edithturn** [comment]
@potiuk, in `pre_commit_breeze_cmd_line.py` I am validating when breeze is not installed.

Some questions I have:
* Breeze in CI should always be installed in the environment, why we should validate the installation for CI?
* One thing I notices is that we don't have instructions to "uninstall breeze" in case the developer wanted to do it.

Please let me know if this is the way to do it or if I have misunderstood this problem.

File: scripts/ci/pre_commit/pre_commit_breeze_cmd_line.py [hunk 1]
```diff
@@ -79,6 +80,19 @@ def verify_all_commands_described_in_docs():
         sys.exit(1)
 
 
+def verify_breeze_installed():
+    result = subprocess.getoutput('breeze version')
```
Comment (uranusjr) [inline scripts/ci/pre_commit/pre_commit_breeze_cmd_line.py]: Maybe use `shutil.which()`? Not as accurate (fails if `breeze` is not “our” command but from somewhere else, for example), but a lot faster.

Comment (potiuk) [reply scripts/ci/pre_commit/pre_commit_breeze_cmd_line.py]: Yeah. Shutil.which is better. It's also portable and it's enough (we don't verify the output of `version` command anyway.

File: scripts/ci/pre_commit/pre_commit_breeze_cmd_line.py [hunk 2]
```diff
@@ -79,6 +80,19 @@ def verify_all_commands_described_in_docs():
...
+    if "Breeze installed from" in result:
+        print_help_for_all_commands()
+        verify_all_commands_described_in_docs()
+    else:
+        print(
+            "You don't have Breeze installed, follow BREEZE.rst to install it \
```
Comment (potiuk) [inline scripts/ci/pre_commit/pre_commit_breeze_cmd_line.py]: we should use rich console and [red] for error and [yellow] for "action" to be done by the user

**Comment by potiuk** [comment]
> * Breeze in CI should always be installed in the environment, why we should validate the installation for CI?

Yeah. It **should**. But it already turned out that it's not always is. See the conversation here: https://github.com/apache/airflow/pull/22961 . We really want people who are not familiar with breeze and CI got precise information about what is wrong. They (even @kaxil !) cannot easily figure out what's wrong when they see a cryptic stack-trace - mostly because they do not have the context. The CI is a living system that changes continuously and we can add new jobs in the future or mistakenly remove stuff. Having a nice message explaining what's wrong AND telling the user what to do is way better than cryptic stack trace.

> * One thing I notices is that we don't have instructions to "uninstall breeze" in case the developer wanted to do it.

Very, very food point - can you add it please (separate PR) :) ? It could be added in the cheatsheeet as well as in the BREEZE.rst.

**Comment by potiuk** [comment]
Added issue for uninstallation instruction: https://github.com/apache/airflow/issues/23005

**Comment by edithturn** [comment]
Thank you for the feedback @potiuk @uranusjr, make sense in adding "precise information about what is wrong". 
Now, how I could test this when `breeze` is not installed in ci, it is already in ci and we can't see the behavior. 
* Should I remove the installation from the static check and make a change in BREEZE.rst?

**Comment by potiuk** [comment]
> * Should I remove breeze installation from the static check step and make a change in BREEZE.rst?

Yep

**Comment by edithturn** [comment]
@potiuk I wasn't able to replicated after the changed I added. https://github.com/apache/airflow/runs/6033018954?check_suite_focus=true
Any other parameter I should take into consideration to replicate it, what I did was:
* Remove the installation of breeze from static check steps
* Add an intentional error in a breeze.rst file 
Also that images that are generated each time I make a change in breeze or dev, are not being generating in localmachine but they are in ci server.

**Comment by edithturn** [comment]
I think now that I know how to uninstall Breeze I can try this 🕺🏼

**Comment by edithturn** [comment]
@potiuk I will test in another branch, bcz in this branch I am not able to reproduce the error. But I saw that in a new branch when Breeze is not installed the not friendly message is shown it.
Let me test this in another branch of this same code after doing a rebase.

**Comment by edithturn** [comment]
@potiuk I will close this PR, I am not able to reproduce it on the local machine, and also it has conflicts that I wasn't able to solve. I created another branch with rebase. Could you guide me on what exactly I should change in Breeze.rst to make the images generate?

**Comment by potiuk** [comment]
No worries. I will take it on - we had some problem with update-breeze-file running in "CI" giving different results so I will take a closer look there.

**Pull request closed without merging**